diff --git a/src/main/scala/org/constellation/snapshotstreaming/Configuration.scala b/src/main/scala/org/constellation/snapshotstreaming/Configuration.scala
--- a/src/main/scala/org/constellation/snapshotstreaming/Configuration.scala
+++ b/src/main/scala/org/constellation/snapshotstreaming/Configuration.scala
@@ -116,7 +116,6 @@ object Configuration {
       c.priorityPeerIds.get(env),
       c.snapshot.size,
       c.feeConfigs.get(env).map(SortedMap.from(_)).getOrElse(SortedMap.empty),
-      c.forkInfoStorage,
       c.lastKryoHashOrdinal,
       c.lastLegacyStateProofOrdinal,
       c.incrementalDelegatedStakingStartingOrdinal,
