diff --git a/src/main/scala/org/constellation/snapshotstreaming/TessellationServices.scala b/src/main/scala/org/constellation/snapshotstreaming/TessellationServices.scala
index 16d8eca..2e0472c 100644
--- a/src/main/scala/org/constellation/snapshotstreaming/TessellationServices.scala
+++ b/src/main/scala/org/constellation/snapshotstreaming/TessellationServices.scala
@@ -107,7 +107,7 @@ object TessellationServices {
                 validationErrorStorage
               )
             val currencySnapshotValidator = CurrencySnapshotValidator
-              .make[F](tessellation3Migration, currencySnapshotCreator, SignedValidator.make[F], None, None)
+              .make[F](currencySnapshotCreator, SignedValidator.make[F], None, None)
             CurrencySnapshotContextFunctions.make(currencySnapshotValidator)
           }
         }
